project-base/app/web/imageResizer.php

```
@@ -66,11 +78,39 @@
 
 function getImageFromUrl(string $url): void
 {
-    $image = file_get_contents($url);
+    $ch = curl_init($url);
+    curl_setopt_array($ch, [
+        CURLOPT_RETURNTRANSFER => true,
+        CURLOPT_FOLLOWLOCATION => true,
+        CURLOPT_TIMEOUT => 5,
+        CURLOPT_CONNECTTIMEOUT => 2,
+        CURLOPT_USERAGENT => 'ImageProxy/1.0',
+    ]);
+    $image = curl_exec($ch);
+    $contentType = curl_getinfo($ch, CURLINFO_CONTENT_TYPE);
+    $statusCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+    curl_close($ch);
+
+    if ($statusCode !== 200 || !$image) {
+        http_response_code(404);
+        exit('Image not found');
+    }
```

let's just throw an exception and handle 404 on the one place (line 75), what do you think?

---

fixed